airflow/timetables/datasets.py

```diff
@@ -52,7 +52,7 @@ def __init__(
         self.periodic = timetable.periodic
         self._can_be_scheduled = timetable._can_be_scheduled
 
-        self.run_ordering = timetable.run_ordering
+        self.run_ordering = ("start_date",)
```

Comment:
If possible, can you add the test case for run ordering?

Reply:
Is this choice intentional? `DatasetTriggeredTimetable` (which regular dataset-triggered DAGs use) use `execution_date` instead.

Reply:
I checked the code for `DatasetTriggeredTimetable` and should inherit the run_order from that class for `DatasetOrTimeSchedule`.

Reply:
@dirrao added the tests